File: Testing/RUnitTestingExamples/patient_count_summary.R
library(dplyr)

#' Summarizes the empanelment volume by practice
#'
#' @param   input_data   A tibble containing at least two columns:
#'                       PracticeNM and MRN
#' @return  A tibble of practices with their patient empanelment volume

patient_count_summary <- function(input_data) {
  input_data %>%
    group_by(PracticeNM) %>%
    summarise(PatientCNT = n())
}

File: Testing/PythonUnitTestingExamples/patient_count_summary.py
from pandera.typing import DataFrame

def patient_count_summary(input_data: DataFrame) -> DataFrame:
    """
    Summarizes the empanelment volume by practice

    Parameters
    ----------
    input_data : DataFrame
        Contains at least two columns: PracticeNM and MRN

    Returns
    -------
    DataFrame
        Summary of practices with their patient empanelment volumes
    """

    patient_count_summary_df = input_data.groupby(["PracticeNM"])["MRN"].agg('count').reset_index().rename(columns = {"MRN": "PatientCNT"})

    return patient_count_summary_df
